src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/OffsetOfTests.cs

```diff
@@ -224,6 +224,20 @@ public void OffsetOf_NoLayoutPoint_ThrowsArgumentException()
             AssertExtensions.Throws<ArgumentException>(null, () => Marshal.OffsetOf<NoLayoutPoint>(nameof(NoLayoutPoint.x)));
         }
 
+        [Fact]
+        public void OffsetOf_Field_Generic()
+        {
+            Assert.Equal((IntPtr)4, Marshal.OffsetOf<Point2<int>>(nameof(Point2<int>.y)));
+            Assert.Equal((IntPtr)8, Marshal.OffsetOf<Point2<ulong>>(nameof(Point2<int>.y)));
+            Assert.Equal((IntPtr)4, Marshal.OffsetOf<Point2<float>>(nameof(Point2<int>.y)));
+            Assert.Equal((IntPtr)8, Marshal.OffsetOf<Point2<double>>(nameof(Point2<int>.y)));
+
+            // [COMPAT] Non-blittable generic types with value-type generic arguments are supported in OffsetOf since they've always been allowed
+            // and it likely doesn't meet the bar to break back-compat.
+            Assert.Equal((IntPtr)1, Marshal.OffsetOf<Point2<char>>(nameof(Point2<int>.y)));
+            Assert.Equal((IntPtr)1, Marshal.OffsetOf<Point2<byte>>(nameof(Point2<int>.y)));
```

Comment:
Can we also add a test case for the non-value-type behaviour?

Reply:
We can add a case for a non-value-type generic, but we can't add a reliable test for an instantiation with a reference type.

Whether or not `Point2<string>` is supported depends on if there's a canonical instantiation (aka is the actual instantiation at runtime `Point2<System.__Canon>` or `Point2<string>`)